guides/05_tabular_data_science_and_plots/creating_a_realtime_dashboard_from_google_sheets.md

```diff
@@ -0,0 +1,147 @@
+
+```py
+import gradio as gr
+
+with gr.Blocks() as demo:
+    gr.Markdown("# 📈 Real-Time Line Plot")
+    with gr.Row():
+        with gr.Column():
+            gr.DataFrame(get_data, every=5)
+        with gr.Column():
+            gr.LinePlot(get_data, every=5, x="Date", y="Sales", y_title="Sales ($ millions)", overlay_point=True, width=500, height=500)
+
+demo.queue().launch()  # Run the demo with queuing enabled
+```
+ 
+And that's it! You have a Dashboard that refreshes every 5 seconds, pulling the data from your Google Sheet.
+
+## Private Google Sheets
+
+For private Google Sheets, the process requires a little more work, but not that much! The key difference is that now, we must authenticate ourselves so to authorize access to the private Google Sheets.
+
+### Authentication
+
+To authenticate ourselves, we'll obtain credentials from Google Cloud. You can do this for free in just a couple of minutes.
+
+1. First, log in to your Google Cloud account and go to the Google Cloud Console (https://console.cloud.google.com/)
+
+2. In the Cloud Console, click on the hamburger menu in the top-left corner and select "APIs & Services" from the menu. If you do not have an existing project, you will need to create one.
+
+3. Then, click the "+ Enabled APIs & services" button, which allows you to enable specific services for your project. Search for "Google Sheets API", click on it, and click the "Enable" button. If you see the "Manage" button, then Google Sheets is already enabled, and you're all set. 
+
+4. In the APIs & Services menu, click on the "Credentials" tab and then click on the "Create credentials" button.
+
+5. In the "Create credentials" dialog, select "Service account key" as the type of credentials to create, and give it a name. **Note down the email of the service account**
+
+6. After selecting the service account, select the "JSON" key type and then click on the "Create" button. This will download the JSON key file containing your credentials to your computer. It will look something like this:
+
+```json
+{
+ "type": "service_account",
+ "project_id": "your project",
+ "private_key_id": "your private key id",
+ "private_key": "private key",
+ "client_email": "email",
+ "client_id": "client id",
+ "auth_uri": "https://accounts.google.com/o/oauth2/auth",
+ "token_uri": "https://accounts.google.com/o/oauth2/token",
+ "auth_provider_x509_cert_url": "https://www.googleapis.com/oauth2/v1/certs",
+ "client_x509_cert_url":  "https://www.googleapis.com/robot/v1/metadata/x509/email_id"
+}
+```
+
+### Querying
+
+Once you have the credentials `.json` file, you can use the following steps to query your Google Sheet:
+
+1. Click on the "Share" button in the top-right corner of the Google Sheet. Share the Google Sheets with the email address of the service from Step 5 of authentication subsection (this step is important!). Then click on the "Get shareable link" button. This will give you a URL that looks something like this:
```

Comment:
there's a weird bug where the numbering doesn't get formatted correctly when separated by codeblocks 

<img width="720" alt="Screen Shot 2022-12-23 at 5 35 37 PM" src="https://user-images.githubusercontent.com/9021060/209344923-0eadb79d-ea62-44f4-b095-55cfc80aee48.png">

instead, to fix this add a backslash before the period 

> `1\.`
> `2\.`
etc

Reply:
Got it, thanks